layers/ip4.go

```diff
@@ -110,9 +112,15 @@ func (ip *IPv4) SerializeTo(b gopacket.SerializeBuffer, opts gopacket.SerializeO
 	}
 	bytes[0] = (ip.Version << 4) | ip.IHL
 	bytes[1] = ip.TOS
-	binary.BigEndian.PutUint16(bytes[2:], ip.Length)
+	switch runtime.GOOS {
```

Comment:
@gconnell You're right. I overlooked it sorry. Over the wire it's in big endian. But when you're working with INET raw sockets in those OSes, the len and frag offset are in little endian. Pretty sure this is only with raw INET sockets. On the wire the byte order is big-endian. I'm not sure why, but it is.

See https://developer.apple.com/library/mac/documentation/Darwin/Reference/ManPages/man4/ip.4.html "Note that the ip_off and ip_len fields are in host byte order."

There is also https://developer.apple.com/library/mac/documentation/Darwin/Reference/ManPages/man4/icmp.4.html
"Incoming packets are received with the IP header and options intact.  Note that the ip_off and ip_len fields are in host byte order"

I've gotta go for now, but I'll be back in 30 minutes to discuss a solution.